### PR TITLE
stages/prompt: add ability to set separate labels and values for choices

### LINKED ISSUE #5197: stages/prompt: Add custom value to RadioButtonGroup and Dropdown
**Is your feature request related to a problem? Please describe.**
Since #4822 we have RadioButtonGroups and Dropdowns in prompts. Sadly we cannot set an internal value.
Value and the choice seen by the user is the same.
Processing the choices is a bit difficult, especially when using an external api.

**Describe the solution you'd like**
The option to have a list of strings where the value and displayed is the same.
As well as a list of tuples which contains value (string) and the displayed value (string).
`return [("value1", "choice1"), ("value2", "choice2")]` -> `<option (...) value="value">choice1</option> <option (...) value="value2">value2</option>`

**Describe alternatives you've considered**
Using only values produces duplicated code or duplicated api calls (getting id for an object after user chose an option) or bad ui/ux.

**Additional context**
important lines: `value="${choice}"` and `<option (...)>${choice}</option>;`
[Source](https://github.com/goauthentik/authentik/blob/main/web/src/flow/stages/prompt/PromptStage.ts#LL149C1-L180C19)
```
            case PromptTypeEnum.Dropdown:
                return `<select class="pf-c-form-control" name="${prompt.fieldKey}">
                    ${prompt.choices
                        ?.map((choice) => {
                            return `<option
                            value="${choice}"
                            ?selected=${prompt.placeholder === choice}
                        >
                            ${choice}
                        </option>`;
                        })
                        .join("")}
                </select>`;
            case PromptTypeEnum.RadioButtonGroup:
                return (
                    prompt.choices
                        ?.map((choice) => {
                            return ` <div class="pf-c-check">
                                    <input
                                        type="radio"
                                        class="pf-c-check__input"
                                        name="${prompt.fieldKey}"
                                        checked="${prompt.placeholder === choice}"
                                        required="${prompt.required}"
                                        value="${choice}"
                                    />
                                    <label class="pf-c-check__label">${choice}</label>
                                </div>
                            `;
                        })
                        .join("") || ""
                );
```

### PR DESCRIPTION
**Pull request by ErikAhlund**
## Details

Fixes #5197

This PR adds support for custom label and values in the select and radio components of the PromptStage by modifying the select object to accept an array of objects with value and label properties, while maintaining backward compatibility with string arrays. Key changes include:

Updated PromptStage.ts to handle both array of strings and array of { value: string, label: string } for select and radio inputs.

Added unit tests in authentik/stages/prompt/tests.py to verify new functionality and edge cases. (Further tests might be needed)

Updated relevant documentation to reflect the new option format in website/docs/add-secure-apps/flows-stages/stages/prompt/index.md

Please let me know if additional changes or tests are needed!

---

## Checklist

-   [ x ] Local tests pass (`ak test authentik/`)
-   [ x ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ x ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ x ] The code has been formatted (`make web`)

If applicable

-   [ x ] The documentation has been updated
-   [ x ] The documentation has been formatted (`make docs`)

### CODE REVIEW

File: website/docs/add-secure-apps/flows-stages/stages/prompt/index.md [hunk 1]
```diff
@@ -65,7 +65,9 @@ A field placeholder, shown within the input field.
...
 will be evaluated as a Python expression. This happens in the same environment as [_Policies_](../../../../customize/policies/expression.mdx).
 
-In the case of `Radio Button Group` and `Dropdown` prompts, this field defines all possible values (choices). When interpreted as-is, only one value will be allowed (the placeholder string). When interpreted as expression, a list of values can be returned to define multiple choices. For example, `return ["first option", 42, "another option"]` defines 3 possible values.
+In the case of `Radio Button Group` and `Dropdown` prompts, this field defines all possible values (choices). When interpreted as-is, only one value will be allowed (the placeholder string). When interpreted as expression, a list of values can be returned to define multiple choices. For example, `return ["first option", 42, {"label": "another option", "value": "some value"}]` defines 3 possible values.
+
+A choice can be represented as a string or other primitive (which will be turned into a string), or as an object with parameters `value` and/or `label`.Sow `return ["Option 1"]` is equivalent to `return [{"label": "Option 1", "value": "Option 1"}]`.
```
Comment (macmoritz) [inline website/docs/add-secure-apps/flows-stages/stages/prompt/index.md]: just a minor typo:  
```suggestion
A choice can be represented as a string or other primitive (which will be turned into a string), or as an object with parameters `value` and/or `label`. So `return ["Option 1"]` is equivalent to `return [{"label": "Option 1", "value": "Option 1"}]`.
```

File: authentik/stages/prompt/stage.py [hunk 2]
```diff
@@ -227,6 +231,18 @@ def get_prompt_challenge_fields(self, fields: list[Prompt], context: dict, dry_r
...
 
+    def clean_choices(self, choices):
+        clean = []
+        for choice in choices:
+            if isinstance(choice, dict):
+                clean.append(
+                    {"value": str(choice.get("value", "")), "label": str(choice.get("label", ""))}
+                )
+            else:
+                clean.append(str(choice))  # Should we force it to be a dict with value + label?
```
Comment (macmoritz) [inline authentik/stages/prompt/stage.py]: IMHO parsing the list to a dict with value und label is cleaner, due to less code and complexity in the frontend.  
But I am not part of authentik, this is just my personal opinion.

Comment (ErikAhlund) [reply authentik/stages/prompt/stage.py]: I agree, that way the data structure received by the frontend is consistent

Comment (ErikAhlund) [reply authentik/stages/prompt/stage.py]: I just applied the changes to parse the list to a dict. I also made a new PromptChoiceSerializer which generates a PromptChoice type for the frontend. Currently it has a mandatory value and label (strings and other primitives are turned into this object by setting the value and label to the primitive)

**Comment by ErikAhlund** [comment]
@BeryJu @tanberry Is there anything I can do to help this PR move along? It's already been a month since I created it, am I missing something?

**Review by GirlBossRush — APPROVED** [review]

**Review by BeryJu — APPROVED** [review]

File: website/docs/add-secure-apps/flows-stages/stages/prompt/index.md [hunk 3]
```diff
@@ -80,7 +82,7 @@ With the `hidden` prompt, the initial value will also be the actual value, becau
 By default, the initial value is interpreted as-is. If you enable _Interpret initial value as expression_, the initial value
 will be evaluated as a Python expression. This happens in the same environment as [_Policies_](../../../../customize/policies/expression.mdx).
 
-In the case of `Radio Button Group` and `Dropdown` prompts, this field defines the default choice. When interpreted as-is, the default choice will be the initial value string. When interpreted as expression, the default choice will be the returned value. For example, `return 42` defines `42` as the default choice.
+In the case of `Radio Button Group` and `Dropdown` prompts, this field defines the default choice. When interpreted as-is, the default choice will be the initial value string. When interpreted as expression, the default choice will be the returned value. For example, `return 42` defines `42` as the default choice. When a choice is defined as an object `{"label": "Option", "value": "internal-value"}` the initial value needs to be set to the value string `internal-value` in this case.
```
Comment (dominic-r) [inline website/docs/add-secure-apps/flows-stages/stages/prompt/index.md]:
```suggestion
In the case of `Radio Button Group` and `Dropdown` prompts, this field defines the default choice. When interpreted as-is, the default choice will be the initial value string. When interpreted as expression, the default choice will be the returned value. For example, `return 42` defines `42` as the default choice. When a choice is defined as an object `{"label": "Option", "value": "internal-value"}`, the initial value needs to be set to the value string `internal-value` in this case.
```

File: website/docs/add-secure-apps/flows-stages/stages/prompt/index.md [hunk 4]
```diff
@@ -65,7 +65,9 @@ A field placeholder, shown within the input field.
...
 will be evaluated as a Python expression. This happens in the same environment as [_Policies_](../../../../customize/policies/expression.mdx).
 
-In the case of `Radio Button Group` and `Dropdown` prompts, this field defines all possible values (choices). When interpreted as-is, only one value will be allowed (the placeholder string). When interpreted as expression, a list of values can be returned to define multiple choices. For example, `return ["first option", 42, "another option"]` defines 3 possible values.
+In the case of `Radio Button Group` and `Dropdown` prompts, this field defines all possible values (choices). When interpreted as-is, only one value will be allowed (the placeholder string). When interpreted as expression, a list of values can be returned to define multiple choices. For example, `return ["first option", 42, {"label": "another option", "value": "some value"}]` defines 3 possible values.
+
+A choice can be represented as a string or other primitive (which will be turned into a string), or as an object with parameters `value` and/or `label`. So `return ["Option 1"]` is equivalent to `return [{"label": "Option 1", "value": "Option 1"}]`.
```
Comment (dominic-r) [inline website/docs/add-secure-apps/flows-stages/stages/prompt/index.md]:
```suggestion
A choice can be a string (or any primitive that will be converted to a string) or an object with `value` and/or `label` properties. For example, `return ["Option 1"]` is equivalent to `return [{"label": "Option 1", "value": "Option 1"}]`.
```

File: website/docs/add-secure-apps/flows-stages/stages/prompt/index.md [hunk 5]
```diff
@@ -65,7 +65,9 @@ A field placeholder, shown within the input field.
 By default, the placeholder is interpreted as-is. If you enable _Interpret placeholder as expression_, the placeholder
 will be evaluated as a Python expression. This happens in the same environment as [_Policies_](../../../../customize/policies/expression.mdx).
 
-In the case of `Radio Button Group` and `Dropdown` prompts, this field defines all possible values (choices). When interpreted as-is, only one value will be allowed (the placeholder string). When interpreted as expression, a list of values can be returned to define multiple choices. For example, `return ["first option", 42, "another option"]` defines 3 possible values.
+In the case of `Radio Button Group` and `Dropdown` prompts, this field defines all possible values (choices). When interpreted as-is, only one value will be allowed (the placeholder string). When interpreted as expression, a list of values can be returned to define multiple choices. For example, `return ["first option", 42, {"label": "another option", "value": "some value"}]` defines 3 possible values.
```
Comment (dominic-r) [inline website/docs/add-secure-apps/flows-stages/stages/prompt/index.md]:
```suggestion
For `Radio Button Group` and `Dropdown` prompts, this field defines the available choices. When used as a plain string, it represents a single allowed value (the placeholder). When used as an expression, it can return a list of choices. For example, `return ["first option", 42, {"label": "another option", "value": "some value"}]` defines three possible values.
```

**Review by dominic-r — APPROVED** [review]
lgtm for docs
